File: C#/0617. Merge Two Binary Trees.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode MergeTrees(TreeNode t1, TreeNode t2) {
        if(t1==null && t2==null){
            return t1;
        }    
        else if(t1!=null && t2==null){
            return t1;
        }
        else if(t1==null && t2!=null){
            return t2;
        }
        else{
            t1.val=t1.val+t2.val;
            t1.left=MergeTrees(t1.left,t2.left);
            t1.right=MergeTrees(t1.right,t2.right);
            return t1;
        }
    }
}

File: archives/20190519python/0617. Merge Two Binary Trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def mergeTrees(self, t1: TreeNode, t2: TreeNode) -> TreeNode:
        if not t1 and not t2:
            return None
        elif t1 and not t2:
            return t1
        elif not t1 and t2:
            return t2
        else:
            root=TreeNode(t1.val+t2.val)
            root.left=self.mergeTrees(t1.left,t2.left)
            root.right=self.mergeTrees(t1.right,t2.right)
            return root
        
